python/cuspatial/cuspatial/geometry/geoseries.py

```diff
@@ -0,0 +1,792 @@
+        """
+        Returns a new GeoPandas GeoSeries object from the coordinates in
+        the cuspatial GeoSeries.
+        """
+        output = []
+        if nullable is True:
+            raise ValueError("cuGeoSeries doesn't support N/A yet")
+        if index is None:
+            index = self.index
+        if isinstance(index, cudf.Index):
+            index = index.to_pandas()
+        output = [geom.to_shapely() for geom in self]
+        return gpGeoSeries(output, index=index)
+
+    def __repr__(self):
+        return f"{self.to_pandas().__repr__()}\n" f"(GPU)\n"
+
+    def _dump(self):
+        return (
+            f"POINTS\n"
+            f"{self.points.__repr__()}\n"
+            f"MULTIPOINTS\n"
+            f"{self.multipoints.__repr__()}\n"
+            f"LINES\n"
+            f"{self.lines.__repr__()}\n"
+            f"POLYGONS\n"
+            f"{self.polygons.__repr__()}\n"
+        )
+
+    def copy(self, deep=True):
+        """
+        Create a copy of all of the GPU-backed data structures in this
+        GeoSeries.
+        """
+        result = GeoSeries([])
+        result._data = self._data
+        result._reader = self._reader
+        result._points = self.points.copy(deep)
+        result._multipoints = self.multipoints.copy(deep)
+        result._lines = self.lines.copy(deep)
+        result._polygons = self.polygons.copy(deep)
+        result.types = self.types
+        result.lengths = self.lengths
+        result.index = self.index
+        return result
+
+    def __iter__(self):
+        self._iter_index = 0
+        return self
+
+    def __next__(self):
+        if self._iter_index >= len(self):
+            raise StopIteration
+        result = self[self._iter_index]
+        self._iter_index = self._iter_index + 1
+        return result
+
+    class GeoSeriesLocIndexer:
+        """
+        Not yet supported.
+        """
+
+        def __init__(self):
+            # Todo: Easy to implement with a join.
+            raise NotImplementedError
+
+    class GeoSeriesILocIndexer:
+        """
+        Each row of a GeoSeries is one of the six types: Point, MultiPoint,
+        LineString, MultiLineString, Polygon, or MultiPolygon.
+        """
+
+        def __init__(self, sr):
+            self._sr = sr
+
+        def __getitem__(self, index):
+            if not isinstance(index, slice):
+                return self._getitem_int(index)
+            else:
+                raise NotImplementedError
+                # This slice functionality is not ready yet
+                # return self._getitem_slice(index)
+
+        def _getitem_int(self, index):
+            type_map = {
+                "p": gpuPoint,
+                "mp": gpuMultiPoint,
+                "l": gpuLineString,
+                "ml": gpuMultiLineString,
+                "poly": gpuPolygon,
+                "mpoly": gpuMultiPolygon,
+            }
+            return type_map[self._sr.types[index]](self._sr, index)
```

Comment:
These are yet another reason to inherit from `cudf.Series`. You can use the parent class's implementation to get either a cudf Scalar or a cudf Series which you could translate zero copy into the geoSeries.